file/builder.go

```diff
@@ -40,7 +41,13 @@ func (b *stateBuilder) build() (*utils.KongRawState, *utils.KonnectRawState, err
 	b.konnectRawState = &utils.KonnectRawState{}
 
 	if b.targetContent.Info != nil {
-		b.selectTags = b.targetContent.Info.SelectorTags
+		if len(b.selectTags) > 0 {
+			if !reflect.DeepEqual(b.selectTags, b.targetContent.Info.SelectorTags) {
```

Comment:
Compare these only if the tags are specified in the state file (there is a workflow possible where tags are only specified as CLI option).
Also, do a proper element-wise comparison rather than reflect, which will throw an error if the ordering is different.

Reply:
3e1a594